src/server/qtquick/wquickcursor.cpp

```diff
@@ -291,7 +297,7 @@ void WQuickCursorPrivate::leaveOutput(WOutput *output)
     if (!cursorSurfaceItem)
         return;
     auto surface = cursorSurfaceItem->surface();
-    if (!surface)
+    if (!surface || surface->isInvalidated())
```

Comment:
There is no need, if handle `aboutToBeInvalidated` signal in WSurfaceContentItem